src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/PersonOldAddressMapper.java

```diff
@@ -41,4 +41,10 @@ private String getAddressLineOrNull(List<StringType> addressLines, int index) {
         }
         return addressLines.get(index).toString();
     }
+
+    @Override
+    public boolean canMap(Parameters parameters) {
+        return ParametersExtension.extractPatient(parameters)
+            .getAddress().size() > 1;
```

Comment:
What would happen for `getAddress().size() > 1`? I think there would data provided in FHIR request that is ignored by the adaptor. Need to reject the request with 400 if anything is missing.

Reply:
It's condition for optional segment. Why would we reject request without optional data?

Reply:
I misunderstood, and a few lines above it even explains why this is 1
.skip(1) // previous address is always the second occurrence of address in the Patient resource